src/client.js

```diff
@@ -24,9 +26,19 @@ var Promise = require('bluebird'),
  *                      given client ID (string)
  *
  * @param   log         Optional bunyan instance for logging
+ * @param   access_token
+ *                      Optional widesky access token.
+ *                      For application whom would like to seed the client
+ *                      with a known access token to avoid having to perform
+ *                      the login operation again.
  */
-var WideSkyClient = function(base_uri, username,
-        password, client_id, client_secret, log) {
+var WideSkyClient = function(base_uri,
+                             username,
+                             password,
+                             client_id,
+                             client_secret,
+                             log,
+                             access_token) {
```

Comment:
I wonder if a better way to do this would be to have a `credentials` object…

* `{"username": <USER>, "password": <PW>, "client_id": <ID>, "client_secret": <SEC>}`: Conventional log-in
* `{"refresh_token": <TOKEN>, "client_id": <ID>, "client_secret": <SEC>}`: Refresh token log-in
* `{"access_token": <TOKEN>}`: Use existing access token

An ES5-compatible and backward-compatible way of decoding this might be:
```javascript
var WideSkyClient = function() {
    var args = Array.prototype.slice.call(arguments);
    /* base_uri always first argument */
    var base_uri = args.shift();
    var credentials = args.shift();
    if (typeof(credentials) === 'string') {
        /* Username/password given */
        credentials = {
            username: credentials,
            password: args.shift(),
            client_id: args.shift(),
            client_secret: args.shift()
        };
    }

    /* etc */
};
```

Otherwise I can see a proliferation of arguments appearing for the different log-in strategies (user account, refresh token, social auth, etc).

Reply:
I've raised BA-640